plugins/outputs/iotdb/iotdb.go

```diff
@@ -0,0 +1,277 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package iotdb
+
+import (
+	_ "embed"
+	"errors"
+	"fmt"
+	"math"
+	"time"
+
+	"github.com/apache/iotdb-client-go/client"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/internal/choice"
+	"github.com/influxdata/telegraf/plugins/outputs"
+)
+
+// DO NOT REMOVE THE NEXT TWO LINES! This is required to embed the sampleConfig data.
+//go:embed sample.conf
+var sampleConfig string
+
+type IoTDB struct {
+	Host            string          `toml:"host"`
+	Port            string          `toml:"port"`
+	User            string          `toml:"user"`
+	Password        string          `toml:"password"`
+	Timeout         config.Duration `toml:"timeout"`
+	ConvertUint64To string          `toml:"uint64_conversion"`
+	TimeStampUnit   string          `toml:"timestamp_precision"`
+	TreatTagsAs     string          `toml:"convert_tags_to"`
+
+	session *client.Session
+	Log     telegraf.Logger `toml:"-"`
```

Comment:
Maybe we should group exported and unexported...
```suggestion
	Log     telegraf.Logger `toml:"-"`

	session *client.Session
```

Reply:
All right! This is a more ideal structure. Thanks.